app/jobs/pageflow/process_file_job.rb

```diff
@@ -5,13 +5,16 @@ class ProcessFileJob < ApplicationJob
     include StateMachineJob
 
     def perform_with_result(file, _options)
-      file.processed_attachment = file.unprocessed_attachment
-      file.save!
+      file.attachment.reprocess!
 
-      :ok
+      if file.valid?
+        :ok
+      else
+        raise ActiveRecord::RecordInvalid
+      end
     rescue ActiveRecord::RecordInvalid, Errno::ENAMETOOLONG
-      file.processed_attachment = nil
-      file.save!
+      file.attachment = nil
+      file.attachment_file_name = 'error'
       :error
```

Comment:
We actually need to remove the attachment if it's invalid. Otherwise the state transition fails due to the attachment validation. But I'll change it so that we keep the `file_name` in order to be able to retry the processing